### PR TITLE
refactor: Move selection to the data_types crate and remove redundant implementations

### PR DESCRIPTION
**Pull request by alamb**
Inspired by @carols10cents 's comments here: https://github.com/influxdata/influxdb_iox/pull/700#pullrequestreview-577487299

I propose removing the duplicated `Selection` struct by putting it in DataTypes (like Schema)

It seems like this is a reasonable (very small) struct to include in `data_types`

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: mutable_buffer/src/chunk.rs [hunk 1]
```diff
@@ -1,11 +1,10 @@
 //! Represents a Chunk of data (a collection of tables and their data within
 //! some chunk) in the mutable store.
-use arrow_deps::datafusion::{error::Result as ArrowResult, logical_plan::LogicalPlan};
 use arrow_deps::{
     arrow::record_batch::RecordBatch,
     datafusion::{
-        error::DataFusionError,
-        logical_plan::{Expr, ExpressionVisitor, Operator, Recursion},
+        error::{DataFusionError, Result as DatafusionResult},
+        logical_plan::{Expr, ExpressionVisitor, LogicalPlan, Operator, Recursion},
```
Comment (carols10cents) [inline mutable_buffer/src/chunk.rs:7]: 👍 👍 👍 👍

File: server/src/db/selection.rs [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-    }
-}
-
-/// Convert a query::Selection into one specific for read buffer
-pub fn to_read_buffer_selection(selection: Selection<'_>) -> ColumnSelection<'_> {
-    match selection {
-        Selection::All => ColumnSelection::All,
-        Selection::Some(cols) => ColumnSelection::Some(cols),
-    }
-}
```
Comment (carols10cents) [inline server/src/db/selection.rs:18]: 👍 👍 👍 👍

**Review by carols10cents — APPROVED** [review]
What a great idea, I love it! ;)

**Comment by alamb** [comment]
@e-dard  -- is this change OK with you (it changes the ReadBuffer slightly)

**Review by e-dard — APPROVED** [review]
LGTM 👍

**Pull request merged**